docs/js-guide.md

```diff
@@ -103,6 +103,36 @@ function getPDFTitle() {
 class PDFIntegration { ... };
 ```
 
+#### Ref naming
+
+Preact does not allow the `ref` prop to be passed verbatim on non-HTML components. Instead it
+offers a wrapper method called `forwardRef` to extract the ref for you. Unfortunately the wrapper
+method is included with `preact/compat` which is a library we currently don't import nor want
+to because of a number of global side effects. For this reason, we simply use a custom property
+with a "Ref" suffix and pass the ref object to that property. For example, `buttonRef`, `panelRef`
+or `fooBarRef` would all be valid names. It is then up to the component to pull out a the ref
+passed in to that prop and attach it to the underlying `HTMLElement`.
+
+e.g.
+
+```js
+/**
+ * @typedef FooBarProps
+ * @prop {import('preact').Ref<HTMLInputElement>} fooBarRef
```

Comment:
The `Ref<T>` type works in the example here. A caveat is that it is the most general possible ref type:

```js
	type RefObject<T> = { current: T | null };
	type RefCallback<T> = (instance: T | null) => void;
	type Ref<T> = RefObject<T> | RefCallback<T>;
```

If the component expects to be able to access the `current` property then it will need to be more specific (eg. by using `{ current: HTMLInputElement }` as the type.

Reply:
I'm unclear exactly what to use in the example. I don't believe we use the general case anywhere in our code base so should we use this an example?